### PR TITLE
[lib] Generalize Control.timeout type.

### PR DESCRIPTION
**Pull request by ejgallego**
Trivial change that I need for other stuff coming soon.

### CODE REVIEW

File: lib/control.mli [hunk 1]
```diff
@@ -16,11 +16,11 @@ val check_for_interrupt : unit -> unit
 (** Use this function as a potential yield function. If {!interrupt} has been
     set, il will raise [Sys.Break]. *)
 
-val timeout : int -> (unit -> 'a) -> exn -> 'a
+val timeout : int -> ('a -> 'b) -> 'a -> exn -> 'b
 (** [timeout n f e] tries to compute [f], and if it fails to do so before [n]
```
Comment (SkySkimmer) [inline lib/control.mli]: Comment needs updating.

File: lib/control.mli [hunk 2]
```diff
@@ -16,11 +16,11 @@ val check_for_interrupt : unit -> unit
...
     seconds, it raises [e] instead. *)
 
-type timeout = { timeout : 'a. int -> (unit -> 'a) -> exn -> 'a }
+type timeout = { timeout : 'a 'b. int -> ('a -> 'b) -> 'a -> exn -> 'b }
 
 val set_timeout : timeout -> unit
```
Comment (SkySkimmer) [inline lib/control.mli]: Why is this exposed?

**Review by SkySkimmer — COMMENTED** [review]
LGTM except for comment update.

File: lib/control.mli (hunk 1 above)
Comment (ejgallego) [reply lib/control.mli]: Thanks.

File: lib/control.mli (hunk 2 above)
Comment (ejgallego) [reply lib/control.mli]: Historical artifact, cleaned up.

**Comment by ppedrot** [comment]
This looks essentially isomorphic to me, but I guess it makes the code more elegant...

File: lib/control.ml [hunk 3]
```diff
@@ -80,12 +80,10 @@ let windows_timeout n f e =
...
-type timeout = { timeout : 'a. int -> (unit -> 'a) -> exn -> 'a }
+type timeout = { timeout : 'a 'b. int -> ('a -> 'b) -> 'a -> exn -> 'b }
 
 let timeout_fun = match Sys.os_type with
 | "Unix" | "Cygwin" -> ref { timeout = unix_timeout }
 | _ -> ref { timeout = windows_timeout }
 
-let set_timeout f = timeout_fun := f
-
 let timeout n f e = !timeout_fun.timeout n f e
```
Comment (SkySkimmer) [inline lib/control.ml]: Since there's no set_timeout there's no need for a ref.

**Comment by ejgallego** [comment]
> This looks essentially isomorphic to me, but I guess it makes the code more elegant...

@ppedrot when command take an state I'll need `timeout` to be of this type.

**Review by ppedrot — APPROVED** [review]
@ejgallego If `↦` stands for a pure arrow, `{ foo : 'a. ('a -> t) ↦ 'a -> t }` and `(unit -> t) -> t` are isomorphic. But anyways, I don't see any harm in merging this PR, so let's approve it so that it gets merged quickly.

**Comment by ejgallego** [comment]
Ah sure, I see what you mean XD. You know I am not a friend of isomorphisms when programming XD

**Comment by ejgallego** [comment]
Merçi !

**Pull request merged**